io.c

```diff
@@ -1459,18 +1460,13 @@ static int io_req_defer(struct io_ring_ctx *ctx, struct io_kiocb *req,
 	if (!io_sequence_defer(ctx, req) && list_empty(&ctx->defer_list))
 		return 0;
 
-	sqe_copy = kmalloc(sizeof(*sqe_copy), GFP_KERNEL);
-	if (!sqe_copy)
-		return -EAGAIN;
-
+	sqe_copy = &req->cached_sqe;
```

Comment:
every req now has a cached copy of the original req. This saves later memory allocation, all memory needs are allocated and initialized upfront.